File: iOS10/LightOrganApp/CircleView.swift
//
//  CircleView.swift
//  LightOrganApp
//
//  Created by Marcin Kruszyński on 07/05/16.
//  Copyright © 2016 Marcin Kruszyński. All rights reserved.
//

import UIKit

@IBDesignable
class CircleView: UIView {
    
    @IBInspectable var circleColor: UIColor = UIColor.red
    
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        configure()
    }
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        configure()
    }
    
    func configure() {
        contentMode = .redraw
    }
    
    override func draw(_ rect: CGRect) {
        drawCircle(circleColor)
    }    
    
    func drawCircle(_ color: UIColor) {
        
        let context = UIGraphicsGetCurrentContext()
        
        let a = min(bounds.size.width, bounds.size.height)
        let leftX = self.bounds.midX - a / 2
        let topY = self.bounds.midY - a / 2
        let rectangle = CGRect(x: leftX, y: topY, width: a, height: a)
        
        context?.setFillColor(circleColor.cgColor)
        context?.fillEllipse(in: rectangle)
    }
}

File: Xamarin.Forms/LightOrganApp/LightOrganApp.iOS/CircleView.cs
using CoreGraphics;
using Foundation;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Text;
using UIKit;

namespace LightOrganApp.iOS
{
    [Register("CircleView"), DesignTimeVisible(true)]
    public class CircleView: UIView
    {
        private UIColor circleColor;

        [Export("CircleColor"), Browsable(true)]
        public UIColor CircleColor
        {
            get { return circleColor; }
            set
            {
                circleColor = value;
                SetNeedsDisplay();
            }
        }        

        public CircleView(IntPtr p): base(p)
        {
            Initialize();
        }

        public CircleView()
        {
            Initialize();
        }

        private void Initialize()
        {
            circleColor = UIColor.Red;
            ContentMode = UIViewContentMode.Redraw;

            SetNeedsDisplay();
        }

        public override void Draw(CGRect rect)
        {
            base.Draw(rect);

            DrawCircle(CircleColor);
        }

        private void DrawCircle(UIColor color)
        {
            using (var context = UIGraphics.GetCurrentContext())
            {
                var a = Math.Min(Bounds.Size.Width, Bounds.Size.Height);
                var leftX = Bounds.GetMidX() - a / 2;
                var topY = Bounds.GetMidY() - a / 2;
                var rectangle = new CGRect(leftX, topY, a, a);

                context.SetFillColor(CircleColor.CGColor);
                context.FillEllipseInRect(rectangle);
            }
        }
    }
}
